scylla-server-rust/src/controller/data_controller.rs

```diff
@@ -0,0 +1,42 @@
+use axum::{
+    extract::{Path, State},
+    Json,
+};
+use serde::Serialize;
+
+use crate::{
+    error::ScyllaError,
+    prisma::{self},
+    services::data_service,
+    Database,
+};
+
+#[derive(Serialize, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct DataSend {
+    pub time: i64,
+    pub values: Vec<String>,
+}
+
+// convert the prisma type to the client type for JSON encoding
+impl From<&prisma::data::Data> for DataSend {
+    fn from(value: &prisma::data::Data) -> Self {
+        DataSend {
+            values: value.values.iter().map(|f| f.to_string()).collect(),
+            time: value.time.timestamp_millis(),
+        }
+    }
+}
+
+pub async fn get_data(
+    State(db): State<Database>,
```

Comment:
why do the controllers need the database? cant the services just get this?

Reply:
the axum/tower/tokio way of doing things is a very seperate state for the lifetime of each route.  Each route is given a context and state (the db is in the Arc so it can contain references across threads) and if it leaves its state anything it modifies can be poisoned without a mutex.  By adhering to this rule, we can ensure we never need to worry about mutexing or RwLocking anything, as axum handles the threads and load balancing for us.